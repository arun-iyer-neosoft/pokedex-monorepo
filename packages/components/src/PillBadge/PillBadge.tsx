import React from 'react';

interface PillBadgeProps {
	children: string;
	background?: 'blue' | 'purple';
}

export const PillBadge = ({ children, background }: PillBadgeProps) => {
	let backgroundColor = background === 'blue' ? 'bg-blue-700' : 'bg-purple-700';
	return (
		<span
			className={`cursor-pointer ${backgroundColor} text-white text-lg rounded-full px-8 py-1 uppercase`}
		>
			{children}
		</span>
	);
};
