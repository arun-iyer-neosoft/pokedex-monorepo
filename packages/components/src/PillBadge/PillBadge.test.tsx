import { render, screen } from '@testing-library/react';
import * as React from 'react';
import { PillBadge } from './PillBadge';

describe('PillBadge', () => {
	it('Renders the Pill with text', () => {
		const { container } = render(<PillBadge>data</PillBadge>);

		expect(screen.getByText('data')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('Renders button with the correct background color', () => {
		const { container } = render(<PillBadge background='blue'>data</PillBadge>);

		expect(screen.getByText('data')).toHaveClass('bg-blue-700');
		expect(container).toMatchSnapshot();
	});
});
