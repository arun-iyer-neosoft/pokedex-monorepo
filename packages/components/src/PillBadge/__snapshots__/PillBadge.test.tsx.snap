// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PillBadge Renders button with the correct background color 1`] = `
<div>
  <span
    class="cursor-pointer bg-blue-700 text-white text-lg rounded-full px-8 py-1 uppercase"
  >
    data
  </span>
</div>
`;

exports[`PillBadge Renders the Pill with text 1`] = `
<div>
  <span
    class="cursor-pointer bg-purple-700 text-white text-lg rounded-full px-8 py-1 uppercase"
  >
    data
  </span>
</div>
`;
